examples/mcmc.jl

```diff
@@ -1,52 +1,100 @@
 using Gen
 using UnicodePlots
+using Distributions
+using Plots
+
+include(joinpath(@__DIR__, "helpers.jl"))
+
+# Truncated Distributions
+struct TruncNorm <: Gen.Distribution{Float64} end
+const trunc_norm = TruncNorm()
+function Gen.random(::TruncNorm, mu::U, noise::T, low::T, high::T) where {U<:Real,T<:Real}
+    d = Distributions.Truncated(Distributions.Normal(mu, noise),
+                                low, high)
+    return Distributions.rand(d)
+end;
+function Gen.logpdf(::TruncNorm, x::Float64, mu::U, noise::T, low::T, high::T) where {U<:Real,T<:Real}
+    d = Distributions.Truncated(Distributions.Normal(mu, noise),
+                                low, high)
+    return Distributions.logpdf(d, x)
+end;
+
+# this function does not work because the max and min function do not offer an interface for the @trace macro, they would have to be implemented as probability distributions
```

Comment:
not super relevant but, in general, deterministic calls should not be in `@trace` expressions. anyhow, this would not serve as a good, simple proposal since it includes additional random moves that aren't present in the generative model (this would require involutive mcmc)